NetSerializer/Primitives.cs

```diff
@@ -318,7 +324,122 @@ public static void ReadPrimitive(Stream stream, out decimal value)
 			value = new Decimal(arr);
 		}
 
-#if NO_UNSAFE
+#if NETCOREAPP
+		public static void WritePrimitive(Stream stream, string value)
+		{
+			if (value == null)
+			{
+				WritePrimitive(stream, (uint)0);
+				return;
+			}
+
+			if (value.Length == 0)
+			{
+				WritePrimitive(stream, (uint)1);
+				return;
+			}
+
+			Span<byte> buf = stackalloc byte[StringByteBufferLength];
+
+			var totalChars = value.Length;
+			var totalBytes = Encoding.UTF8.GetByteCount(value);
+
+			WritePrimitive(stream, (uint)totalBytes + 1);
+			WritePrimitive(stream, (uint)totalChars);
+
+			var totalRead = 0;
+			ReadOnlySpan<char> span = value;
+			for (;;)
+			{
+				var finalChunk = totalRead + totalChars >= totalChars;
+				Utf8.FromUtf16(span, buf, out var read, out var wrote, isFinalBlock: finalChunk);
+				stream.Write(buf.Slice(0, wrote));
+				totalRead += read;
+				if (read >= totalChars)
+				{
+					break;
+				}
+
+				span = span[read..];
+				totalChars -= read;
+			}
+		}
+
+		private static readonly SpanAction<char, (int, Stream)> _stringSpanRead = StringSpanRead;
```

Comment:
Any reason for this variable? Why not just use StringSpanRead directly?

Reply:
I believe Roslyn is not smart enough to cache the delegate so if you used it directly it would result in a delegate allocation every time.

Reply:
Ok. Can you add a short comment to the code about it?

Reply:
Done.